include/mxnet/tuple.h

```diff
@@ -220,6 +237,7 @@ class Tuple {
    * \return the ostream
    */
   friend std::ostream &operator<<(std::ostream &os, const Tuple<ValueType> &t) {
+    if (t.ndim() == -1) return os;
```

Comment:
Why not we print something like "?"?

Reply:
How about making it more specific: `UNKNOWN_SHAPE`?